include/net/aws_fota.rst

```diff
@@ -3,58 +3,105 @@
 
-MQTT is used to receive notification that an update is available, and to retrieve metadata about the update.
-HTTP is used to download the update payload.
+* The other device has valid (but different) certificates that use the same AWS IoT policy as the original device.
+* The other device is subscribed to the same MQTT topic as the original device.
 
-It is up to the application that uses the library to restart when the FOTA is complete.
+The library uses MQTT to receive notification that an update is available, and to retrieve metadata about the update.
+It uses HTTP to download the update payload.
+
+It is up to the application that uses the library to restart the device when the FOTA is complete.
+
+The AWS FOTA library is used in the :ref:`aws_fota_sample` sample.
 
 Configuration
 *************
 
-Use Kconfig to configure the MQTT payload buffer sizes and the buffers used to store the version string, host name, and file path.
+Configure the following parameters when using this library:
+
+- :option:`CONFIG_AWS_FOTA_PAYLOAD_SIZE`
+- :option:`CONFIG_AWS_FOTA_VERSION_STRING_MAX_LEN`
+- :option:`CONFIG_AWS_FOTA_HOSTNAME_MAX_LEN`
+- :option:`CONFIG_AWS_FOTA_FILE_PATH_MAX_LEN`
 
 
 Implementation
 **************
 
-The implementation uses the job document shown below for passing information from AWS Jobs to the device:
+The following sequence diagram shows how a firmware over-the-air update is implemented through the use of `AWS IoT MQTT`_, `AWS IoT jobs`_, and `AWS Simple Storage Service (S3)`_.
+
+.. figure:: /images/aws_fota_dfu_sequence.svg
+   :alt: AWS FOTA sequence diagram for doing FOTA through AWS jobs
+
+
+AWS S3 server
+=============
+
+The firmware files for download must be stored in a bucket on an AWS S3 server.
+You can use an AWS instance that is already set up, such as `nRF Cloud`_, or set up your own bucket.
+
+When setting up your own bucket, make sure to configure the permissions as shown in the following screenshot:
+
+.. figure:: /images/aws_s3_bucket_permissions.png
+   :alt: Bucket permissions in AWS S3
+
+To update the permissions for an existing bucket, select your bucket and navigate to **Permissions** > **Block public access**.
+
+In addition to the permissions, you must configure a bucket policy.
+To determine a suitable security scheme for your application, see `AWS S3 Developer Guide: Using Bucket Policies and User Policies`_ and `AWS S3 Developer Guide: Bucket Policy Examples`_.
+To configure the policy, select your bucket and navigate to **Permissions** > **Bucket Policy**.
+
+For testing purposes, you can use the following, very permissive, bucket policy (replace *bucket_name* with the name of your bucket):
 
 .. code-block:: javascript
 
+   {    "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": "*",
+                "Action": "s3:GetObject",
+                "Resource": "arn:aws:s3:::*bucket_name*/\*"
```

Comment:
is the `\` for escape in rst?

Reply:
I missed the correct markup, but yes, now it will show up as:
"Resource": "arn:aws:s3:::_bucket_name_/*"